factcast-snapshotcache-mongodb/src/main/java/org/factcast/core/snap/mongo/MongoDbSnapshotCache.java

```diff
@@ -0,0 +1,225 @@
+/*
+ * Copyright © 2017-2020 factcast.org
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.factcast.core.snap.mongo;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.mongodb.ReadConcern;
+import com.mongodb.ReadPreference;
+import com.mongodb.TransactionOptions;
+import com.mongodb.WriteConcern;
+import com.mongodb.client.ClientSession;
+import com.mongodb.client.MongoClient;
+import com.mongodb.client.MongoCollection;
+import com.mongodb.client.MongoDatabase;
+import com.mongodb.client.gridfs.GridFSBucket;
+import com.mongodb.client.gridfs.GridFSBuckets;
+import com.mongodb.client.model.*;
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
+import javax.validation.constraints.NotNull;
+import lombok.NonNull;
+import lombok.Setter;
+import lombok.extern.slf4j.Slf4j;
+import org.bson.Document;
+import org.bson.types.ObjectId;
+import org.factcast.factus.serializer.SnapshotSerializerId;
+import org.factcast.factus.snapshot.SnapshotCache;
+import org.factcast.factus.snapshot.SnapshotData;
+import org.factcast.factus.snapshot.SnapshotIdentifier;
+
+@SuppressWarnings("deprecation")
+@Slf4j
+public class MongoDbSnapshotCache implements SnapshotCache {
+  public static final String PROJECTION_CLASS_FIELD = "projectionClass";
+  public static final String AGGREGATE_ID_FIELD = "aggregateId";
+  public static final String SNAPSHOT_SERIALIZER_ID_FIELD = "snapshotSerializerId";
+  public static final String LAST_FACT_ID_FIELD = "lastFactId";
+  public static final String FILE_ID_FIELD = "fileId";
+  public static final String EXPIRE_AT_FIELD = "expireAt";
+
+  // Recommended by the docs to use Majority for gridfs operations
+  public static final TransactionOptions txnOptions =
+      TransactionOptions.builder()
+          .readPreference(ReadPreference.primary())
+          .readConcern(ReadConcern.MAJORITY)
+          .writeConcern(WriteConcern.MAJORITY)
+          .build();
+
+  private final MongoDbSnapshotProperties properties;
+  private final MongoCollection<Document> collection;
+  private final MongoClient mongoClient;
+
+  @VisibleForTesting @Setter private GridFSBucket gridFSBucket;
```

Comment:
```suggestion
  @Setter(onMethod=@__({@VisibleForTesting})) private GridFSBucket gridFSBucket;
```

This way the annotation is placed on the generated method, not on the field (which is private anyway).

Reply:
forget about this, we do it the way you wrote it basically everywhere in factcast :)